examples/providers/examples/connect_builtin.rs

```diff
@@ -0,0 +1,52 @@
+//! Example of using the `RootProvider<N, T: BoxTransport>::connect_builtin` to create a provider
+//! from a connection string. The connection string can be a HTTP, WS or IPC endpoint.
+
+use alloy::node_bindings::Anvil;
+use alloy_network::Ethereum;
+use alloy_provider::{Provider, RootProvider};
+use alloy_transport::BoxTransport;
+use eyre::Result;
+use futures_util::StreamExt;
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let anvil = Anvil::new().block_time(1).try_spawn()?;
+    let http = anvil.endpoint();
+    let ws = anvil.ws_endpoint();
+
+    // Instantiate a HTTP transport provider by passing the http endpoint url
+    let http_provider =
+        RootProvider::<Ethereum, BoxTransport>::connect_builtin(http.as_str()).await?;
```

Comment:
More of a general comment and a personal opinion but I'm not a fan of the `connect_builtin` name. I don't think it is particularly clear to users and sounds like an internal method rather than the equivalent of a `default`.

Reply:
Possibly beyond the scope of the intended PR but I would also prefer to see the parameter being strongly typed (`Url`-like) instead of a `&str`